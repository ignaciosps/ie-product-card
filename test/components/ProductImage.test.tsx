import React from 'react';
import renderer from 'react-test-renderer';
import { ProductCard, ProductImage } from '../../src/components';
import { product2 } from '../data/products';

describe('ProductImage', () => {
  test('Debe de mostrar el componente correctamente la imagen personalizada', () => {
    const wraper = renderer.create(
      <ProductImage img={product2.img} className="custom-class" />
    );
    expect(wraper.toJSON()).toMatchSnapshot();
  });
  test('Debe de mostrar el componente con la imagen del producto', () => {
    const wraper = renderer.create(
      <ProductCard product={product2}>{() => <ProductImage />}</ProductCard>
    );
    expect(wraper.toJSON()).toMatchSnapshot();
  });
});
