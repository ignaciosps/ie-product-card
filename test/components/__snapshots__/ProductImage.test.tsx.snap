// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage Debe de mostrar el componente con la imagen del producto 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Coffee Mug - Con Imagen"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`ProductImage Debe de mostrar el componente correctamente la imagen personalizada 1`] = `
<img
  alt="No Description"
  className="productImg custom-class"
  src="./coffee-mug.png"
/>
`;
